mlrun/api/utils/builder.py

```diff
@@ -424,6 +496,18 @@ def build_image(
 
         runtime.spec.clone_target_dir = path.join(tmpdir, "mlrun", relative_workdir)
 
+    from mlrun.api.crud import Secrets
+
+    provider = mlrun.common.schemas.SecretProviderName.kubernetes
+    project_secrets = (
+        Secrets()
+        .list_project_secrets(
+            project=project, provider=provider, allow_secrets_from_k8s=True
+        )
+        .secrets
+    )
```

Comment:
There is duplication here. see `_generate_builder_env` there we get the secrets straight from k8s which is practically the same as this since the provider is k8s which means you will be adding all project secrets twice and will explode in validation

Reply:
thanks, I'll remove this (It will not explode in validation cause it's only comparing the extra args with the existing, but it is a duplication).